### PR TITLE
Modify the examples to add restore keys

### PR DESCRIPTION
**Pull request by fadnavistanmay**
Modify the examples to add restore keys

Fixes [AB#1638428](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1638428)

### CODE REVIEW

File: docs/pipelines/caching/index.md [hunk 1]
```diff
@@ -243,12 +248,13 @@ If you use `PackageReferences` to manage NuGet dependencies directly within your
...
-  NUGET_PACKAGES: $(Pipeline.Workspace)/.nuget/packages
+  PACKAGES_PATH: $(Pipeline.Workspace)/.nuget/packages
 
 steps:
 - task: CacheBeta@0
   inputs:
-    key: nuget | packages.lock.json
+    key: nuget | $(Agent.OS) | packages.lock.json
+    restoreKeys: nuget | $(gent.OS)
     path: $(NUGET_PACKAGES)
```
Comment (johnterickson) [inline docs/pipelines/caching/index.md]: Typo

**Comment by fadnavistanmay** [comment]
Hi John, I have modified some examples to include multiple lines, Thanks.

File: docs/pipelines/caching/index.md [hunk 2]
```diff
@@ -78,6 +78,8 @@ steps:
 - task: CacheBeta@0
   inputs:
     key: yarn | $(Agent.OS) | yarn.lock
+    restoreKeys: yarn | $(Agent.OS)
```
Comment (johnterickson) [inline docs/pipelines/caching/index.md]: I don't think this is valid YAML

```
restoreKeys: | 
  yarn | $(Agent.OS)
  yarn
```

Comment (fadnavistanmay) [reply docs/pipelines/caching/index.md]: Yeah, changed it.

**Comment by steved0x** [comment]
@fadnavistanmay Is this PR ready for merging? If so let me know and I will get it merged. Thank you!

**Comment by fadnavistanmay** [comment]
@johnterickson - could you please take a look. Have added the restore keys in examples, plus the quotes.. Thanks.

File: docs/pipelines/caching/index.md [hunk 3]
```diff
@@ -176,7 +183,8 @@ steps:
 
 - task: CacheBeta@0
   inputs:
-    key: ccache | $(Agent.OS)
+    key: 'ccache | "$(Agent.OS)"'
+    restoreKeys: ccache
```
[the remark this answers is not in the document]
Comment (fadnavistanmay) [reply docs/pipelines/caching/index.md]: You mean, just for cpp cache, to not share across OS?

File: docs/pipelines/caching/index.md [hunk 4]
```diff
@@ -77,7 +77,10 @@ variables:
 steps:
 - task: CacheBeta@0
   inputs:
-    key: yarn | $(Agent.OS) | yarn.lock
+    key: 'yarn | "$(Agent.OS)" | yarn.lock'
+    restoreKeys: |
+      'yarn | "$(Agent.OS)"'
```
[the remark this answers is not in the document]
Comment (fadnavistanmay) [reply docs/pipelines/caching/index.md]: Changing it.

File: docs/pipelines/caching/index.md [hunk 5]
```diff
@@ -243,12 +255,15 @@ If you use `PackageReferences` to manage NuGet dependencies directly within your
 
 ```yaml
 variables:
-  NUGET_PACKAGES: $(Pipeline.Workspace)/.nuget/packages
+  PACKAGES_PATH: $(Pipeline.Workspace)/.nuget/packages
```
[the remark this answers is not in the document]
Comment (fadnavistanmay) [reply docs/pipelines/caching/index.md]: ahh, NUGET_PACKAGES require an absolute path, hence the change of variable, trying to find the link from where I stumbled upon this.

[the remark this answers is not in the document]
Comment (fadnavistanmay) [reply docs/pipelines/caching/index.md]: nowwwww I remembered, that was just a random variable I used 🤦‍♂ - and I changed the restore task definition to modify it.

**Comment by PRMerger6** [comment]
@fadnavistanmay : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/pipelines/caching/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fadnavistanmay) [reply docs/pipelines/caching/index.md]: reverting this.

**Comment by PRMerger6** [comment]
@fadnavistanmay : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by fadnavistanmay** [comment]
Hi @steved0x - could you please merge this. Thanks.

**Pull request merged**